
# HG changeset patch
# User Thomas Jollans <thomas@jollans.com>
# Date 1270586274 -7200
# Node ID d7404c713e9d2dabdd2e8198bff708c7f6523a8f
# Parent  1c8390a60a3ea7e53e110ca428d158c6ea016933
Display large artwork/icon and small icon in the notification (on KDE4, at least...)

diff -r 1c8390a60a3e -r d7404c713e9d src/xmms2tray/__init__.py
--- a/src/xmms2tray/__init__.py	Mon Apr 05 15:55:36 2010 +0200
+++ b/src/xmms2tray/__init__.py	Tue Apr 06 22:37:54 2010 +0200
@@ -125,8 +125,8 @@
         v = res.value()
         
         if isinstance(v, numbers.Integral):
-	    # this is the ID. I want the whole info.
-	    self.xmms.medialib_get_info(res.value(),self.newsong)
+            # this is the ID. I want the whole info.
+            self.xmms.medialib_get_info(res.value(),self.newsong)
         else:
             if isinstance(v, basestring): # coverart
                 info = self.curinfo
@@ -138,7 +138,7 @@
                 coverimg = None
                 self.curinfo = info
 
-            img = os.getcwd() + '/data/xmms2_64.png'
+            img = self._logo_img
             if HAVE_IMAGING and 'picture_front' in info:
                 if info['picture_front'] == self.lastimg[0]:
                     #same image, just use last.
@@ -178,10 +178,14 @@
                 nowpl = info['url'].split('/')[-1].replace('+', ' ')
             self.set_tip()
             if self.config['notify'] and HAVE_NOTIFY:
-                n = pynotify.Notification(_('Now Playing'), nowpl, img)
+                if self._notify_id != info['id']:
+                    self._notify_id = info['id']
+                    self._notification = pynotify.Notification(_('Now Playing'), nowpl, self._logo_img)
+                pbuf = gtk.gdk.pixbuf_new_from_file(img)
+                self._notification.set_icon_from_pixbuf(pbuf)
                 if self.config['att_note']:
-                    n.set_property('status-icon', self.icon)
-                n.show()
+                    self._notification.set_property('status-icon', self.icon)
+                self._notification.show()
 
     def songchange(self, res):
         if self.curinfo['id'] == res.value():
@@ -293,6 +297,9 @@
 
         if HAVE_NOTIFY:
             pynotify.init("XMMS2tray")
+            self._notify_id = 0
+            self._notification = None
+            self._logo_img = os.getcwd() + '/data/xmms2_64.png'
         self.xmms.playback_status(self.pbstatus_cb)
         self.xmms.broadcast_playback_status(self.pbstatus_cb)
         self.xmms.playback_current_id(self.newsong)

